File: baymax's_shopping_trip-15/solution.py
#!/usr/bin/python

total = raw_input()

for i in xrange(0, int(total)):
    input = raw_input().split(" ")
    m = int(input[2])
    current = int(input[0])/int(input[1])
    free = current
    while free >= m:
        tmp = free/m
        current += tmp
        free = tmp
    print current

File: baymax's_shopping_trip-15/solution.cpp
#include <iostream>

using namespace std;

int main() {
  int t, n, c, m, current, free, tmp;
  cin >> t;
  for(int i = 0; i < t; i++){
    cin >> n >> c >> m;
    current = n/c;
    free = current;
    while(free >= m){
      tmp = free/m;
      current += tmp;
      free = tmp;
    }
    cout << current << endl;
  }
  return 0;
}